app/controllers/oauth/applications_controller.rb

```diff
@@ -109,12 +121,57 @@ def set_user
     end
 
     def app_params
-      params.require(:doorkeeper_application).permit(
-        :name, :redirect_uri, :scopes, :email_subject_prefix, :lead_application_source,
-        :email_from_address, :confidential,
-        :can_access_private_user_data, :can_find_or_create_accounts, :can_message_users,
-        :can_skip_oauth_screen,
-      )
+      if @user.is_administrator?
+        params.require(:doorkeeper_application).permit(
+          :name, :redirect_uri, :scopes, :email_subject_prefix, :lead_application_source,
+          :email_from_address, :confidential,
+          :can_access_private_user_data, :can_find_or_create_accounts, :can_message_users,
+          :can_skip_oauth_screen
+        )
+      elsif @user.oauth_applications.include?(@application)
+        params.require(:doorkeeper_application).permit(
+          :redirect_uri
+        )
+      end
+    end
+
+    def add_application_owners
+      member_ids = validated_member_ids
+      return false if @application.errors.any?
+      return true if !current_user.is_administrator?
+    
+      @application.owner.update_attributes(member_ids: member_ids)
+    end
+
+    def validated_member_ids
+      return [] if !params[:member_ids].present?
+    
+      if !params[:member_ids].match(SPACE_SEPARATED_NUMBERS_REGEX)
+        @application.errors.add(:owner, 'Member Ids must be a space separated list of integers')
+        return false
+      end
+    
+      member_ids = params[:member_ids].split.map(&:to_i)
+      member_ids.each do |member|
+        if !User.where(id: member).exists?
+          @application.errors.add(:owner, "#{member} is not a valid user id")
+          return false
+        end
+      end
+      return member_ids.unshift(@application.owner.owner_ids[0])
```

Comment:
Should this be code that is in the group logic?  Seems not the best for outsider users to have to remember to do this.

If the code has to live here, can you add a comment?  Future readers will not know what is happening.

Reply:
I had a comment, but it got lost in the other code changes. It is added back. I don;t think this belongs in the group since it is specific to editing the member_ids.